src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -120,7 +118,19 @@ private[actions] abstract class RefreshActionBase(
             _) =>
           location
             .allFiles()
-            .map(f => FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
+            .map { f =>
+              // For each file, if it already has a file id, add that id to its corresponding
+              // FileInfo. Note that if the content of an existing file is changed, it still
+              // keeps its id. This will not cause an issue as the file will show as
+              // deleted/appended and its index entries will be fixed during refresh.
+              val filePath = f.getPath.toString
+              previousIndexLogEntry.fileNameToIdMap.get(filePath) match {
+                case Some(id) =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime, id)
+                case _ =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime)
```

Comment:
Can we prefer using built-in Option's functions if it doesn't hurt the readability?

For example:
```scala
previousIndexLogEntry.fileNameToIdMap.get(filePath)
                .map(FileInfo(filePath, f.getLen, f.getModificationTime, _))
                .getOrElse(FileInfo(filePath, f.getLen, f.getModificationTime))
```